wagtail_ab_testing/models.py

```diff
@@ -34,16 +39,68 @@ class Variant(models.TextChoices):
 
     page = models.ForeignKey('wagtailcore.Page', on_delete=models.CASCADE, related_name='ab_tests')
     name = models.CharField(max_length=255)
+    hypothesis = models.TextField(blank=True)
     treatment_revision = models.ForeignKey('wagtailcore.PageRevision', on_delete=models.CASCADE, related_name='+')
     goal_event = models.CharField(max_length=255)
     # TODO Page chooser
     goal_page = models.ForeignKey('wagtailcore.Page', null=True, blank=True, on_delete=models.SET_NULL, related_name='+')
     sample_size = models.PositiveIntegerField()
+    created_by = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.SET_NULL, null=True, blank=True, related_name='+')
     status = models.CharField(max_length=20, choices=Status.choices, default=Status.DRAFT)
     winning_variant = models.CharField(max_length=9, null=True, choices=Variant.choices)
+    first_started_at = models.DateTimeField(null=True)
+
+    # Because an admin can pause/resume tests, we need to make sure we record the amount of time it has been running
+    previous_run_duration = models.DurationField(default=timedelta(0))
+    current_run_started_at = models.DateTimeField(null=True)
 
     objects = AbTestManager()
 
+    def get_goal_event_display(self):
+        """
+        Returns the display name of the goal event.
+        """
+        for event_type_slug, event_type in EVENT_TYPES.items():
+            if event_type_slug == self.goal_event:
+                return event_type.name
+
+        return self.goal_event
+
+    def start(self):
+        """
+        Starts/unpauses the test.
+        """
+        if self.status in [self.Status.DRAFT, self.Status.PAUSED]:
+            self.status = self.Status.RUNNING
+            self.current_run_started_at = timezone.now()
+
+            if self.status == self.Status.DRAFT:
+                self.first_started_at = self.current_run_started_at
+
+            self.save(update_fields=['status', 'current_run_started_at', 'first_started_at'])
+
+    def pause(self):
+        """
+        Pauses the test.
+        """
+        if self.status == self.Status.RUNNING:
+            self.status = self.Status.PAUSED
+            self.previous_run_duration += timezone.now() - self.current_run_started_at
+            self.current_run_started_at = None
+
+            self.save(update_fields=['status', 'previous_run_duration', 'current_run_started_at'])
+
+    def total_running_duration(self):
+        """
+        Returns the total duration that this test has been running.
+        """
+        duration = self.previous_run_duration
+
+        if self.status == self.Status.RUNNING:
+            duration += timezone.now() - self.current_run_started_at
```

Comment:
I'm curious, instead of working with timezones to get the duration of a campaign, could we use UTC _all the time_? Im just thinking.. it might be easier (if possible, that is) to work with a standardized time rather than timezones.

Reply:
The timezone should be saved into the database along with the time itself, so this should be as good as a UTC time.

I used UTC time in the log parts because I need to be able to split the dates and hours in a consistent way (ie not worry about DST)